src/components/Redemptions/RedemptionsPage.tsx

```diff
@@ -0,0 +1,291 @@
+    proposals.forEach((proposal) => {
+      proposal.gpRewards.forEach((reward: any) => {
+        if (reward.beneficiary === currentAccountAddress) {
+          if (reward.tokensForStaker && Number(reward.tokensForStakerRedeemedAt) === 0) {
+            genReward.iadd(new BN(reward.tokensForStaker));
+          }
+          if (reward.daoBountyForStaker && Number(reward.daoBountyForStakerRedeemedAt) === 0) {
+            genReward.iadd(new BN(reward.daoBountyForStaker));
+          }
+          if ((reward.reputationForVoter && Number(reward.reputationForVoterRedeemedAt) === 0)
+              || (reward.reputationForProposer && Number(reward.reputationForProposerRedeemedAt) === 0)) {
+            reputationRewardDaos.add(proposal.dao.id);
+          }
+        }
+      });
+
+      // Add ContributionReward redemptions
+      const contributionReward = proposal.contributionReward;
+      if (contributionReward && contributionReward.beneficiary === currentAccountAddress) {
+        ethReward.iadd(new BN(contributionReward.ethReward));
+
+        if (Number(contributionReward.reputationReward) > 0) {
+          reputationRewardDaos.add(proposal.dao.id);
+        }
+
+        if (contributionReward.externslTokenReward && !contributionReward.externalTokenReward.isZero()) {
+          if (externalTokenRewards[contributionReward.externalToken]) {
+            externalTokenRewards[contributionReward.externalToken].iadd(new BN(contributionReward.externalTokenReward));
+          } else {
+            externalTokenRewards[contributionReward.externalToken] = new BN(contributionReward.externalTokenReward);
+          }
+        }
+      }
+
+    });
+
+    const totalRewards: any[] = [];
+    if (!ethReward.isZero()) {
+      totalRewards.push(formatTokens(ethReward, "ETH"));
+    }
+    if (!genReward.isZero()) {
+      totalRewards.push(formatTokens(genReward, "GEN"));
+    }
+    Object.keys(externalTokenRewards).forEach((tokenAddress) => {
+      totalRewards.push(formatTokens(externalTokenRewards[tokenAddress], tokenSymbol(tokenAddress)));
+    });
+    if (reputationRewardDaos.size > 0) {
+      totalRewards.push(<span>reputation in {reputationRewardDaos.size}&nbsp;DAOs</span>);
+    }
+
+    return <strong>
+      {totalRewards.reduce((acc, v) => {
+        return acc === null ? <React.Fragment>{v}</React.Fragment> : <React.Fragment>{acc} <em>&amp;</em> {v}</React.Fragment>;
+      }, null)}
+    </strong>;
+  }
+}
+
+interface IStateProps extends RouteComponentProps<any> {
+  currentAccountAddress: string;
+}
+
+const mapStateToProps = (state: IRootState) => {
+  return {
+    currentAccountAddress: state.web3.currentAccountAddress,
+  };
+};
+
+interface IDispatchProps {
+  redeemProposal: typeof arcActions.redeemProposal;
+}
+
+const mapDispatchToProps = {
+  redeemProposal: arcActions.redeemProposal,
+};
+
+class FallibleRedemptionsPage extends React.Component<IStateProps & IDispatchProps, null> {
```

Comment:
Fallible?

Reply:
I thought it would make sense to make a wrapping component (FallibleRedemptionsPage) to deal with the data fetching, which can possibly fail, and an inner component (RedemptionsPage), which shouldn't.

Is this a question/remark about the name or about the abstraction? Happy to change either if you have an idea.

Reply:
Oh just the name. What about Redemption Data? Or RedemptionaPageData?

Reply:
I've changed the name in the last rebase to `SubscribedRedemptionsPage` to make it consistent with other components. That's better, isn't it?